src/Npgsql/NpgsqlCommand.cs

```diff
@@ -1447,6 +1513,29 @@ protected override void Dispose(bool disposing)
 
         #region Misc
 
+        NpgsqlStatement TruncateStatementsToOne()
+        {
+            switch (_statements.Count)
```

Comment:
> When would this hit anything but 0 or 1 (given the check for postitional + > 0 statement index up in the parser?)

This happens when you execute an NpgsqlCommand with legacy batching (so now `_statements.Length > 1`). If you then reuse that same NpgsqlCommand instance to execute a single statement, we truncate the statement list.

> how exactly do you construct a 0 statement command?

If you mean when `_statements.Count` is 0 - that's just when a command is initially created (so on first execution). Otherwise I don't think it's possible to actually execute a 0-statement command - CommandText has to be non-empty...